invenio_rdm_records/records/systemfields/draft_status.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+# Copyright (C) 2020 Northwestern University.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Draft status field.
+
+The DraftStatus is used to calculate the status of the draft based on its
+associated review request.
+
+For instance:
+
+.. code-block:: python
+
+    class Record():
+        draft_status = DraftStatus()
+
+"""
+
+from flask_babelex import gettext as _
+from invenio_records.systemfields import SystemField
+
+from invenio_rdm_records.services.errors import ReviewStateError
+
+
+class DraftStatus(SystemField):
```

Comment:
I think we need to factor in both the record and draft type as well (``new draft``, ``edit draft``,  ``next draft``) or we need some other field specifying the draft type.